evmrpc/send.go

```diff
@@ -22,34 +22,41 @@ func NewSendAPI(tmClient rpcclient.Client, txConfig client.TxConfig) *SendAPI {
 	return &SendAPI{tmClient: tmClient, txConfig: txConfig}
 }
 
-func (s *SendAPI) SendRawTransaction(ctx context.Context, input hexutil.Bytes) (common.Hash, error) {
-	tx := new(ethtypes.Transaction)
-	if err := tx.UnmarshalBinary(input); err != nil {
-		return common.Hash{}, err
-	}
-	txdata, err := ethtx.NewTxDataFromTx(tx)
-	if err != nil {
-		return common.Hash{}, err
+func (s *SendAPI) SendRawTransaction(ctx context.Context, input hexutil.Bytes) (hash common.Hash, err error) {
+	var txData ethtx.TxData
+	associateTx := ethtx.AssociateTx{}
+	if associateTx.Unmarshal(input) == nil {
```

Comment:
is the idea here that we try to unmarshall the tx as an associateTx first, and if that fails, fallback to EVMTx?

Reply:
@philipsu522 yes, since we can't (also probably don't want to) make `AssociateTx` part of `ethtypes.Transaction` in `go-ethereum` (and thus can't just deserialize it to `ethtypes.Transaction` and check its type), we'll have to potentially deserialize twice